persist/fs/read.go

```diff
@@ -318,5 +466,24 @@ func (r *reader) EntriesRead() int {
 }
 
 func (r *reader) Close() error {
-	return r.dataFdWithDigest.Close()
+	for i := 0; i < len(r.indexEntriesByOffset); i++ {
+		r.indexEntriesByOffset[i].ID = nil
+	}
+	r.indexEntriesByOffset = r.indexEntriesByOffset[:0]
+
+	if err := munmap(r.indexMmap); err != nil {
+		return err
```

Comment:
Might be better semantics to collect all errors and return once all operations have been attempted.

Reply:
Sure thing.